File: packages/fast-sync-core/src/core/abstraction/data/base_repository.ts
export interface IBaseRepository<T> {
    add(entity: T): Promise<T>;
    update(id: string, entity: T): Promise<T | null>;
    findById(id: string): Promise<T | null>;
    getAll(): Promise<T[]>;
    query(filter: any): Promise<T[]>;
    count(): Promise<number>;
    dispose(): void;
}

File: fast_sync_dart_client/fast_sync_client/lib/src/absraction/data/base_repository.dart
abstract class IBaseRepository<T> {
  Future<T> add(T entity);
  Future<T> update(String id, T entity);
  Future<T?> findById(String id);
  Future<List<T>> getAll();
  Future<List<T>> query(bool Function(T) query);
  Future<int> count();
  void dispose();
}
